core/src/avm1/activation.rs

```diff
@@ -1201,9 +1202,21 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
         Ok(FrameControl::Continue)
     }
 
+    fn url_to_https(&self, url: &str) -> Result<String, Error<'gc>> {
+        if !self.context.https_context || url.is_empty() || !url.contains("://") {
+            Ok(url.to_string())
+        } else {
+            let mut url = Url::parse(url).unwrap();
+            if url.scheme() == "http" {
+                url.set_scheme("https").unwrap();
```

Comment:
Let's avoid the `unwraps` here; either pass the failure upward with `?` or return the URL as-is on failure.

Reply:
I had trouble successfully using ? as I couldn't get map_err to return the right error type. That was the intention.

That said, it seems I might be moving the code from here anyway.